manim/utils/hashing.py

```diff
@@ -233,22 +234,22 @@ def default(self, obj: Any):
         # Serialize it with only the type of the object. You can change this to whatever string when debugging the serialization process.
         return str(type(obj))
 
-    def _cleaned_iterable(self, iterable: Iterable[Any]):
+    def _cleaned_iterable(self, iterable: Iterable[Any]) -> list[Any] | dict[Any, Any]:
```

Comment:
Despite the name, `iterable` can be either a `Sequence` or a `dict`, according to the code. Currently, `iterable` can't be really an `Iterable`, because `_iter_check_list` calculates its length, something which not all iterables have (like map, range or filter).

The name of this parameter (and function) should probably be changed.

Now, we could simply type `iterable` as `Sequence[Any] | dict[Any, Any]`, but may I suggest the following overloads to indicate that, if `iterable` is a `Sequence`, the function returns a `list` and, if it's a `dict`, it returns a `dict`:
```suggestion
    @overload
    def _cleaned_iterable(self, iterable: Sequence[Any]) -> list[Any]: ...
    
    @overload
    def _cleaned_iterable(self, iterable: dict[Any, Any]) -> dict[Any, Any]: ...
    
    def _cleaned_iterable(self, iterable):
```